addons/account_peppol/models/res_company.py

```diff
@@ -48,3 +148,23 @@ def _compute_peppol_purchase_journal_id(self):
                 )
             else:
                 company.peppol_purchase_journal_id = False
+
+    @api.depends('email')
+    def _compute_account_peppol_contact_email(self):
+        for company in self:
+            if not company.account_peppol_contact_email:
+                company.account_peppol_contact_email = company.email
+
+    # -------------------------------------------------------------------------
+    # LOW-LEVEL METHODS
+    # -------------------------------------------------------------------------
+
+    def write(self, vals):
+        for company in self:
+            if not vals.get('peppol_endpoint') or company.peppol_eas != '0208':
+                continue
+            cbe_match = re.search('[0-9]{10}', vals['peppol_endpoint'])
+            if bool(cbe_match):
+                vals['peppol_endpoint'] = cbe_match.group(0)
+
+        return super().write(vals)
```

Comment:
- you need to do the same for create
- you need to take care if someone is writting peppol_eas too
- 
```suggestion
def _sanitize_peppol_endpoint():
    self.ensure_one()
    endpoint = self.peppol_endpoint
    eas = self.peppol_eas

    if not endpoint or not eas:
        return

    if self.country_code == 'BE' and eas == '0208':
        cbe_match = re.search('[0-9]{10}', endpoint)
        if bool(cbe_match):
            self.peppol_endpoint = cbe_match.group(0)

@api.model_create_multi
def create(self, vals_list):
    companies = super().create(vals_list)
    for company in companies:
        company._sanitize_peppol_endpoint()
    return companies

def write(self, vals):
    result = super().write(vals)
    for company in self:
        company._sanitize_peppol_endpoint()
    return result
```